packages/input-selection/test/change.test.ts

```diff
@@ -0,0 +1,97 @@
+    const changeBundles: Cardano.Value[] = [{ coins: 5_000_000n }, { coins: 5_000_000n }, { coins: 4_000_000n }];
+    const result = coalesceChangeBundlesForMinCoinRequirement(changeBundles, computeMinimumCoinQuantity);
+
+    expect(result).toBeDefined();
+    expect(result?.length).toBe(2);
+
+    expect(result![0].coins).toBe(9_000_000n);
+    expect(result![1].coins).toBe(5_000_000n);
+  });
+
+  it('when the middle bundle has less than min ADA coin, coalesce it with the last', async () => {
+    const changeBundles: Cardano.Value[] = [
+      { coins: 10_000_000n },
+      { assets: new Map([[TOKEN1_ASSET_ID, 2333n]]), coins: 7_000_000n },
+      { coins: 5_000_000n }
+    ];
+
+    const result = coalesceChangeBundlesForMinCoinRequirement(changeBundles, computeMinimumCoinQuantity);
+
+    expect(result).toBeDefined();
+    expect(result?.length).toBe(2);
+
+    expect(result![0].coins).toBe(12_000_000n);
+    expect(result![0].assets!.get(TOKEN1_ASSET_ID)).toBe(2333n);
+    expect(result![1].coins).toBe(10_000_000n);
+  });
+
+  it('when the first bundle has less than min ADA coin, coalesce it with the last', async () => {
+    const changeBundles: Cardano.Value[] = [
+      { assets: new Map([[TOKEN1_ASSET_ID, 2333n]]), coins: 7_000_000n },
+      { coins: 5_000_000n },
+      { coins: 5_000_000n }
+    ];
+
+    const result = coalesceChangeBundlesForMinCoinRequirement(changeBundles, computeMinimumCoinQuantity);
+
+    expect(result).toBeDefined();
+    expect(result?.length).toBe(2);
+
+    expect(result![0].coins).toBe(12_000_000n);
+    expect(result![0].assets!.get(TOKEN1_ASSET_ID)).toBe(2333n);
+    expect(result![1].coins).toBe(5_000_000n);
+  });
+
+  it('when the three bundle have less than min ADA coin, coalesce them together', async () => {
+    const changeBundles: Cardano.Value[] = [
+      { assets: new Map([[TOKEN1_ASSET_ID, 2333n]]), coins: 7_000_000n },
+      { assets: new Map([[TOKEN1_ASSET_ID, 2333n]]), coins: 7_000_000n },
+      { assets: new Map([[TOKEN1_ASSET_ID, 2333n]]), coins: 7_000_000n }
+    ];
+
+    const result = coalesceChangeBundlesForMinCoinRequirement(changeBundles, computeMinimumCoinQuantity);
+
+    expect(result).toBeDefined();
+    expect(result?.length).toBe(1);
+
+    expect(result![0].coins).toBe(21_000_000n);
+    expect(result![0].assets!.get(TOKEN1_ASSET_ID)).toBe(6999n);
+  });
+
+  it('when coalescing the three bundles do not reach the min ADA coin return undefined', async () => {
+    const changeBundles: Cardano.Value[] = [
+      { assets: new Map([[TOKEN1_ASSET_ID, 2333n]]), coins: 2_000_000n },
+      { assets: new Map([[TOKEN2_ASSET_ID, 2333n]]), coins: 2_000_000n },
+      { assets: new Map([[TOKEN3_ASSET_ID, 2333n]]), coins: 2_000_000n }
+    ];
+
+    const result = coalesceChangeBundlesForMinCoinRequirement(changeBundles, computeMinimumCoinQuantity);
+
+    expect(result).toBeUndefined();
+  });
```

Comment:
Great work, @AngelCastilloB . 
I would propose adding a test for empty change list?

Reply:
added test here: [734f6a8](https://github.com/input-output-hk/cardano-js-sdk/pull/633/commits/734f6a8b978be63ad8889a7bec2bad3ab7138bd4)